frontend/spec/features/agents_spec.rb

```diff
@@ -0,0 +1,1728 @@
+      end
+
+      it "displays the agent in the agent's index page" do
+        agent = create(:agent_person)
+        run_index_round
+
+        visit '/agents?filter_term[]={"primary_type":"agent_person"}&sort=create_time+desc'
+
+        elements = all('#tabledSearchResults tbody tr')
+        expect(elements.length > 0).to eq true
+        expect(elements[0]).to have_text agent.names.first['primary_name']
+      end
+
+      it 'returns agents in search results and shows their types correctly' do
+        agent = create(:agent_person)
+        run_index_round
+
+        element = find('#global-search-box')
+        element.fill_in with: agent.names.first['primary_name']
+        find('#global-search-button').click
+
+        elements = all('#tabledSearchResults tbody tr')
+        expect(elements.length).to eq 1
+        expect(elements[0]).to have_text 'Person'
+        expect(elements[0]).to have_text agent.names.first['primary_name']
+      end
+
+      it 'will not delete a corp entity agent linked to a repo' do
+        repository = create(:repo, repo_code: "agents_test_#{Time.now.to_i}")
+
+        run_index_round
+
+        element = find('#global-search-box')
+        element.fill_in with: repository.repo_code
+        find('#global-search-button').click
+
+        click_on 'Edit'
+        click_on 'Delete'
+
+        within '#confirmChangesModal' do
+          click_on 'Delete'
+        end
+
+        element = find('.alert.alert-danger.with-hide-alert')
+        expect(element.text).to eq "This agent is linked to a repository and can't be removed."
+      end
+    end
+  end
+
+  describe 'Light Agent Record' do
+    before(:all) do
+      @corpοrate_agent_full = create(:json_agent_corporate_entity_full_subrec)
+      @corpοrate_agent_basic = create(:agent_corporate_entity)
+
+      @repository = create(:repo, repo_code: "light_agent_record_test_#{Time.now.to_i}")
+
+      @data_entry_user = create_user(@repository => ['repository-advanced-data-entry'])
+    end
+
+    before(:each) do
+      login_user(@data_entry_user)
+      select_repository(@repository)
+      visit '/agents/agent_person/new'
+    end
+
+    it 'displays agent_record_identifiers in form' do
+      expect(page).to have_css '#agent_person_agent_record_identifier'
+    end
+
+    it 'hides agent_record_control from form' do
+      expect(page).to_not have_css '#agent_person_agent_record_control'
+    end
+
+    it 'hides agent_other_agency_codes from form' do
+      expect(page).to_not have_css '#agent_person_agent_other_agency_codes'
+    end
+
+    it 'hides agent_conventions_declarations from form' do
+      expect(page).to_not have_css '#agent_person_agent_conventions_declaration'
+    end
+
+    it 'hides agent_maintenance_histories from form' do
+      expect(page).to_not have_css '#agent_person_agent_maintenance_history'
+    end
+
+    it 'hides agent_other_agency_codes from form' do
+      expect(page).to_not have_css '#agent_person_agent_sources'
```

Comment:
The selector here should read `'#agent_person_agent_other_agency_codes'`.

Reply:
This is actually a duplicate example that I will remove. But you are right there that it needs to check that no '#agent_person_agent_other_agency_codes' are there, which is actually already done three examples above. @brianzelip